src/Elasticsearch.Net/Connection/HandlerTracking/RequestDataHttpClientFactory.cs

```diff
@@ -0,0 +1,254 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if DOTNETCORE
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Xml;
+using Elasticsearch.Net;
+
+namespace Elasticsearch.Net
+{
+	/// <summary>
+	/// Heavily modified version of DefaultHttpClientFactory, repurposed for RequestData
+	/// </summary>
+	internal class RequestDataHttpClientFactory : IDisposable
```

Comment:
Can probably just call it `HttpClientFactory`?

Reply:
I renamed it to signal it's been re-purposed specifically for `RequestData`. It's not the same as `DefaultHttpClientFactory`. 

Keen to keep the name less generic.